src/Robot/MathematicianMonad.hs

```diff
@@ -0,0 +1,76 @@
+{-# LANGUAGE DeriveGeneric #-}
+
+module Robot.MathematicianMonad where
+
+import Robot.AutomationData
+import Robot.History
+import Robot.TableauFoundation
+
+import Control.Monad.State
+import GHC.Generics
+import Data.Aeson (FromJSON, ToJSON)
+
+-- | The state we have in the monad is called MathematicianState
+data MathematicianState = MathematicianState
+    { returnFreshName :: Int,
+      -- ^ An integer larger than all the names of free variables.
```

Comment:
I agree with this, but just wanted to mention that arguably we could instead store the names which are in use, so that we can keep using the 'smallest' available name, rather than the smallest name which hasn't been used. Though now that I think about it, this might make the history a bit odd. In any case, I agree what you currently have is the right choice